File: 2-AD-HOC/1091/bee1091.cpp
// 1091 - Divisão da Nlogônia
#include <iostream>

int main(void) {

    int n;
    
    do{

        std::cin>>n;

        if(n != 0) {

            int pX, pY;
            std::cin>>pX>>pY;

            for(int i = 0; i < n; i++) {

                int rX, rY;
                std::cin>>rX>>rY;
    
                if(rX == pX || rY == pY) std::cout<<"divisa"<<std::endl;
                else if(rX > pX) {
                    if(rY > pY) std::cout<<"NE"<<std::endl;
                    else std::cout<<"SE"<<std::endl;
                } else {
                    if(rY > pY) std::cout<<"NO"<<std::endl;
                    else std::cout<<"SO"<<std::endl;
                }
            }
        }

    }while(n != 0);
}

File: 2-AD-HOC/1091/bee1091.java
// 1091 - Divisão da Nlogônia
import java.io.IOException;
import java.util.Scanner;

class bee1091 {

    public static void main(String[] args){
        Scanner scan = new Scanner(System.in);

        int n;

        do{
            n = scan.nextInt();

            if(n != 0) {

                int pX, pY;
                pX = scan.nextInt(); pY = scan.nextInt();

                int rX, rY;
                for(int i = 0; i < n; i++) {
                    rX = scan.nextInt(); rY = scan.nextInt();

                    if(rX == pX || rY == pY) System.out.println("divisa");
                    else if(rX > pX) {
                        if(rY > pY) System.out.println("NE");
                        else if(rY < pY) System.out.println("SE");
                    }
                    else if(rX < pX) {
                        if(rY > pY) System.out.println("NO");
                        else if(rY < pY) System.out.println("SO");
                    }
                }
            }

        }while(n != 0);

        scan.close();
    }
}